### PR TITLE
Create civiform footer

### LINKED ISSUE #7377: Implement shared footer
#### Describe the solution presented in this issue.

Implement shared header and footer, as described in the [PRD](https://docs.google.com/document/d/1VHdYTojiqRiBl4R0LbGsNuSZzOlvgjcWwNnZlLZJJHM/edit?tab=t.0).

(PRD not yet complete, issue tracking that: https://github.com/civiform/civiform/issues/7249)

Note once we have this, there is at least one test (applicant_application_test) which can be augmented to take advantage of being able to log in / log out.

#### Additional Context
Add any other context, link related issues, etc. 

#### Done when:
What criteria need to be met for the issue to be finished? 
- [ ] Clicking button A takes the user to Page B
- [ ] Criteria 2 
- [ ] Criteria 3

### PR DESCRIPTION
**Pull request by greg-hyde**
### Description

Add a basic footer to all pages

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### User visible changes

- [x] Followed steps to [internationalize](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#application-strings) any new strings
  - [x] Added context strings to new [messages](https://github.com/civiform/civiform/blob/main/server/conf/i18n/messages)
  - [x] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)
- [x] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

#### New Features

- [x] Conditioned new functionality on a [FeatureFlag](https://github.com/civiform/civiform/wiki/Feature-Flags)
- [x] Wrote browser tests with the feature flag off and on, etc.

### Issue(s) this completes

Fixes #7377

### CODE REVIEW

**Comment by greg-hyde** [comment]
* **#7953** <a href="https://app.graphite.dev/github/pr/civiform/civiform/7953?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @greg-hyde and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: server/app/assets/stylesheets/uswds/_uswds-theme-custom-styles.scss [hunk 1]
```diff
@@ -119,3 +119,35 @@ See the radio-option below as an example.
 h1.usa-step-indicator__heading {
   line-height: inherit;
 }
+
+/**
+ * USWDS color utils are turned off, so we use the USWDS color manually.
+ * See #7951 for detils
+ */
+.cf-footer-help-links a {
+  color: #1b1b1b;
```
Comment (ssandbekkhaug) [inline server/app/assets/stylesheets/uswds/_uswds-theme-custom-styles.scss:128]: Is there a standard color that should be referenced here?

File: server/app/assets/stylesheets/uswds/_uswds-theme-custom-styles.scss [hunk 2]
```diff
@@ -119,3 +119,35 @@ See the radio-option below as an example.
...
+  .cf-footer-help-bar {
+    padding: 0;
+  }
+
+  .cf-footer-help-links {
+    width: 100%;
+  }
+
+  .cf-footer-help-links a {
+    border-bottom: 1px solid #71767a;
```
Comment (ssandbekkhaug) [inline server/app/assets/stylesheets/uswds/_uswds-theme-custom-styles.scss:145]: Ditto

File: server/conf/i18n/messages [hunk 3]
```diff
@@ -11,7 +11,21 @@
 button.logout=Logout
 label.selectLanguage=Please select your preferred language.
 button.untranslatedSubmit=Submit
+# Text on a link that gives more information about CiviForm. "CiviForm" is the app name and should
+# not be translated.
+footer.aboutCiviform=About CiviForm
 footer.supportLinkDescription=For technical support, contact
+# Text in the footer. The placeholder is a link call to action to visit a website.
+footer.lookingForMoreInformation=Looking for more information and services? {0}
```
Comment (ssandbekkhaug) [inline server/conf/i18n/messages:19]: This is never used anywhere. Is that intentional?

File: server/conf/i18n/messages [hunk 4]
```diff
@@ -11,7 +11,21 @@
...
+# Text in the footer telling users that this is an official website of the specified Civic entity.
+footer.officialWebsiteOf=An official website of the {0}
+# Text on a link that will scroll to the top of the page
+footer.returnToTop=Return to top
+# A message in the footer directing users to technical support. The placeholder is a link to
+# send an email to CiviForm technical support.
+footer.technicalSupport=For technical support: {0}
+# Text within a link to visit the given website
+footer.visitLink=Visit {0}
```
Comment (ssandbekkhaug) [inline server/conf/i18n/messages:26]: Ditto

File: server/conf/i18n/messages (hunk 3 above)
Comment (greg-hyde) [reply server/conf/i18n/messages:19]: good catch it was supposed to be but we don't actually have the info needed for the placeholder value, so we're holding off for now. Filed https://github.com/civiform/civiform/issues/7959 for that chunk of work

File: server/app/assets/stylesheets/uswds/_uswds-theme-custom-styles.scss (hunk 2 above)
Comment (greg-hyde) [reply server/app/assets/stylesheets/uswds/_uswds-theme-custom-styles.scss]: done

File: server/app/assets/stylesheets/uswds/_uswds-theme-custom-styles.scss (hunk 1 above)
Comment (greg-hyde) [reply server/app/assets/stylesheets/uswds/_uswds-theme-custom-styles.scss]: done

File: server/conf/i18n/messages (hunk 4 above)
Comment (ssandbekkhaug) [reply server/conf/i18n/messages:26]: Is `visitLink` going to be used anywhere? If not, let's delete it

**Review by ssandbekkhaug — APPROVED** [review]

File: server/conf/i18n/messages (hunk 4 above)
Comment (greg-hyde) [reply server/conf/i18n/messages]: done

**Pull request merged**